Update CI Visibility Java onboarding instructions (#17907)
DataDog/documentation: content/en/continuous_integration/tests/java.md

{{% /tab %}}
{{% tab "Gradle" %}}

> Make sure to set the `DD_TRACER_VERSION` environment variable to the tracer version you have previously downloaded, and `DD_TRACER_FOLDER` variable to the path where you have downloaded the tracer.

Reviewer: ```suggestion
Make sure to set the `DD_TRACER_VERSION` environment variable to the tracer version you have previously downloaded, and the `DD_TRACER_FOLDER` variable to the path where you have downloaded the tracer.
```

> Make sure to set the `DD_TRACER_VERSION` environment variable to the tracer version you have previously downloaded, and `DD_TRACER_FOLDER` variable to the path where you have downloaded the tracer.

Reviewer: ```suggestion
Make sure to set the `DD_TRACER_VERSION` environment variable to the tracer version you have previously downloaded, and the `DD_TRACER_FOLDER` variable to the path where you have downloaded the tracer.
```

* Test names and durations.
* Predefined environment variables set by CI providers.
* Git commit history including the hash, message, author information, and files changed (without file contents).
* Source code information: relative paths to sources of test classes, line numbers of test methods.
* Information from the CODEOWNERS file.

## Troubleshooting

### The tests are not appearing in Datadog after enabling CI Visibility in the tracer

Ensure that you are using the latest version of the tracer.

> Verify that your build system and testing framework are supported by CI Visibility (refer to the list of supported build systems and test frameworks).

Reviewer: ```suggestion
Verify that your build system and testing framework are supported by CI Visibility. See the list of [supported build systems and test frameworks](#compatibility).
```

Ensure that the `dd.civisibility.enabled` property is set to `true` in the tracer arguments.

> Check the build output for any errors that indicate tracer misconfiguration, such as an unset DD API KEY.

Reviewer: ```suggestion
Check the build output for any errors that indicate tracer misconfiguration, such as an unset `DD_API_KEY` environment variable.
```

### Tests or source code compilation fails when building a project with the tracer attached

By default, CI Visibility runs Java code compilation with a compiler plugin attached.

The plugin is optional, as it only serves to reduce the performance overhead.

Depending on the build configuration, adding the plugin can sometimes disrupt the compilation process.

If the plugin interferes with the build, disable it by adding `dd.civisibility.compiler.plugin.auto.configuration.enabled=false` to the list of `-javaagent` arguments.

### Tests fail when building a project with the tracer attached

In some cases attaching the tracer can break tests, especially if they run asserts on the internal state of the JVM or instances of third-party libraries' classes.

> While the best approach is such cases is to update the tests, there is also a quicker option of disabling the tracer's third-party libraries integrations.

Reviewer: ```suggestion
While the best approach is such cases is to update the tests, there is also a quicker option of disabling the tracer's third-party library integrations.
```

> While the best approach is such cases is to update the tests, there is also a quicker option of disabling the tracer's third-party libraries integrations.

Reviewer: ```suggestion
While the best approach is such cases is to update the tests, there is also a quicker option of disabling the tracer's third-party library integrations.
```
